nexus/deployment/src/blueprint_builder.rs

```diff
@@ -329,6 +455,118 @@ impl<'a> BlueprintBuilder<'a> {
         Ok(Ensure::Added)
     }
 
+    pub fn sled_ensure_zone_nexus(
+        &mut self,
+        sled_id: Uuid,
+    ) -> Result<Ensure, Error> {
+        // If there's already a Nexus zone on this sled, do nothing.
+        let has_nexus = self
+            .parent_blueprint
+            .omicron_zones
+            .get(&sled_id)
+            .map(|found_zones| {
+                found_zones.zones.iter().any(|z| z.zone_type.is_nexus())
+            })
+            .unwrap_or(false);
+        if has_nexus {
+            return Ok(Ensure::NotNeeded);
+        }
+
+        // Whether Nexus should use TLS and what the external DNS servers it
+        // should use are currently provided at rack-setup time, and should be
+        // consistent across all Nexus instances. We'll assume we can copy them
+        // from any other Nexus zone in our parent blueprint.
+        //
+        // TODO-correctness Once these properties can be changed by a rack
```

Comment:
I'm not sure these settings need to be part of policy. They are really customer configuration, and while they could live in policy I'm not sure if it makes sense to put them there or to keep them in a separate customer configuration.

The way I've been thinking of the reconfigurator is as a control system. The policy, inventory, current database state, and current blueprint all feed into the planner to generate a new blueprint which then triggers changes which feeback into the planner. With this, we can have any number of different inputs, and we don't have to force them all into the policy, inventory, or blueprint. However, I'm unsure if that is helpful or if it simplifies things to move items used by the planner into either policy, blueprint, or inventory.

With all that being said, I don't think we can rely on the prior blueprint for this information. It can be configured by an operator and should be read by the planner directly from the database in its "raw" form, or as part of the updated policy. We'll likely also have to update other existing nexus zones, right?

You should also open an issue for this so we don't lose track of it.

Reply:
> I'm not sure these settings need to be part of policy. They are really customer configuration, and while they could live in policy I'm not sure if it makes sense to put them there or to keep them in a separate customer configuration.

I've been thinking of the policy as the customer configuration. It's currently "the sleds that should be part of the rack" and (as of this PR) "the service external IP pool", both of which are provided at RSS and only modified by an operator. But I don't feel super strongly about what we call it and whether parts should live in something else.

> With all that being said, I don't think we can rely on the prior blueprint for this information. It can be configured by an operator and should be read by the planner directly from the database in its "raw" form, or as part of the updated policy. We'll likely also have to update other existing nexus zones, right?

I agree in the limit, but it can't be configured by an operator today, so there's nothing to read from the database. Definitely fine on opening an issue; I think moving the storage of these into the DB (and eventually making them operator-adjustable) shouldn't be rolled into this PR. Are you okay with this as-is for now?

Reply:
Sorry, I didn't mean to give you the impression I wanted you to make changes related to this on this PR. I was just riffing on design for the future. This is good as is for now.

Reply:
Yeah, I think we've been using "policy" to mean any kind of configuration that would be input to planning -- i.e., anything other than the current state or previous blueprint.  Agreed that in an ideal world, this would be configurable by the customer, and so it would be stored somewhere in the database, and we'd use that, and that absent that, we should just look at what other Nexus instances have set here.